lib/src/facts/openbsd/networking_resolver.cc

```diff
@@ -0,0 +1,69 @@
+#include <internal/facts/openbsd/networking_resolver.hpp>
+#include <internal/util/bsd/scoped_ifaddrs.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+#include <facter/util/directory.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <sys/sockio.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+#include <net/if_dl.h>
+#include <netinet/in.h>
+
+using namespace std;
+using namespace facter::util;
+using namespace facter::util::bsd;
+using namespace facter::execution;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    bool networking_resolver::is_link_address(sockaddr const* addr) const
+    {
+        return addr && addr->sa_family == AF_LINK;
+    }
+
+    uint8_t const* networking_resolver::get_link_address_bytes(sockaddr const* addr) const
+    {
+        if (!is_link_address(addr)) {
+            return nullptr;
+        }
+        sockaddr_dl const* link_addr = reinterpret_cast<sockaddr_dl const*>(addr);
+        if (link_addr->sdl_alen != 6) {
+            return nullptr;
+        }
+        return reinterpret_cast<uint8_t const*>(LLADDR(link_addr));
+    }
+
+    boost::optional<uint64_t> networking_resolver::get_link_mtu(string const& interface, void* data) const
```

Comment:
`get_link_mtu` implementation will now be shared between `linux` and `openbsd`; `get_link_address_bytes` and `is_link_address` will be shared between `osx` and `openbsd`. Might be worth pushing the shared implementations up and letting the outliers override them, rather than copying implementations around.

Currently we never instantiate the `bsd::networking_resolver` directly anywhere, we inherit from it for Linux and Mac OS X. I'd vote for pushing the implementations necessary to run on OpenBSD up to BSD, and let Linux and Mac deviate where necessary. That would mean pushing up the Linux implementation for `get_link_mtu`, and the Mac implementations for every other function being overridden here.